src/components/tooltip/tooltip.module.scss

```diff
@@ -17,42 +17,56 @@ $triangle-color: var(--rp-ui-base-almost-black-light);
 
 .tooltip {
   transition: opacity 300ms ease;
-  margin: $DEFAULT_MARGIN;
+  z-index: 9;
+
   &[data-placement*='bottom'] {
-    padding-top: $ARROW_HEIGHT;
-    z-index: 2;
+    margin-top: $DEFAULT_MARGIN;
   }
 
   &[data-placement*='top'] {
-    padding-bottom: $ARROW_HEIGHT;
+    margin-bottom: $DEFAULT_MARGIN;
+
+    .tooltip-content {
+      transform: translate(0, calc(-100% - 8px));
+    }
```

Comment:
I'm not sure if this is a reason, but looks like now dynamic positioning is broken with your changes
With your changes:
![image](https://github.com/user-attachments/assets/5c567600-1af3-425e-9e25-3c21c5c0eedb)
Without your changes:
![image](https://github.com/user-attachments/assets/9d92d111-8184-441e-8eac-45b5f412fa56)

Could you consider [Floating UI arrow](https://floating-ui.com/docs/arrow) or the approach used in Popover component?

Reply:
Done.